mri/reconstructors/single_channel.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This implements the single channel reconstruction.
+"""
+
+from .base import ReconstructorBase
+from ..operators import GradSynthesis, GradAnalysis, WaveletN
+
+
+class SingleChannelReconstructor(ReconstructorBase):
+    """ This class implements the Single channel MR image Reconstruction.
+    For the Analysis case, finds the solution  for x of:
+        (1/2) * sum(||F x - y||^2_2, 1) + mu * ||W x||_1
+
+    For the Synthesis case, finds the solution of:
+        (1/2) * sum(||F Wt alpha - y||^2_2, 1) + mu * ||alpha||_1
+
+    Parameters
+    ----------
+    fourier_op: object of class FFT, NonCartesianFFT or Stacked3DNFFT in
+                mri.operators
+        Defines the fourier operator F in the above equation.
+    linear_op: object, (optional, default None)
+        Defines the linear sparsifying operator W. This must operate on x and
+        have 2 functions, op(x) and adj_op(coeff) which implements the
+        operator and adjoint operator. For wavelets, this can be object of
+        class WaveletN or WaveletUD2 from mri.operators .
+        If None, sym8 wavelet with nb_scale=3 is chosen.
+    regularizer_op: operator, (optional default None)
+        Defines the regularization operator for the regularization function H.
+        If None, the  regularization chosen is Identity and the optimization
+        turns to gradient descent.
+    gradient_formulation: str between 'analysis' or 'synthesis',
+        default 'synthesis'
+        defines the formulation of the image model which defines the gradient.
+    lips_calc_max_iter: int, default 10
+        Defines the maximum number of iterations to calculate the lipchitz
+        constant
+    num_check_lips: int, default 10
+        Number of iterations to check if the lipchitz constant is correct
+    lipschitz_cst: int, default None
```

Comment:
Lispchitz constant is not an int it's a float, I might have missed this one in the above reconstructors